pkgs/by-name/qu/qucsator-rf/package.nix

```
@@ -0,0 +1,48 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  bison,
+  flex,
+  dos2unix,
+  gperf,
+  adms,
+  withAdms ? false,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "qucsator-rf";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "ra3xdh";
+    repo = "qucsator_rf";
+    rev = version;
+    hash = "sha256-IvB4CTvK6x4wwUXMoXIqBku1Hh9em6ITTpwFllYsTEg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    flex
+    bison
+    gperf
+    dos2unix
+  ];
+
+  buildInputs = lib.optionals withAdms [ adms ];
+
+  cmakeFlags = [
+    "-DBISON_DIR=${bison}/bin"
+    "-DWITH_ADMS=${if withAdms then "ON" else "OFF"}"
```

```suggestion
    (lib.cmakeBool "WITH_ADMS" withAdms)
```
You could also use cmakeFeature for the other flag